packages/client-sdk-nodejs/src/internal/pubsub-client.ts

```diff
@@ -244,7 +253,7 @@ export class PubsubClient extends AbstractPubsubClient {
         serviceError.code === Status.INTERNAL &&
         serviceError.details === PubsubClient.RST_STREAM_NO_ERROR_MESSAGE;
       const momentoError = new TopicSubscribe.Error(
-        cacheServiceErrorMapper(serviceError)
+        this.cacheServiceErrorMapper.convertError(serviceError)
```

Comment:
We are not throwing here because we might retry the subscription? Do we want to throw when its the first subscription message?

Reply:
this is a good question, I probably need to look into this part a little more closely.

Reply:
The thing here is that our JS topic subscriptions are callback-based already. So what that means is that there is no thread of execution to try to throw the exception into; if an error occurs during a subscription then the onError callback gets called.

I can't immediately think of a behavior change that the ThrowOnErrors flag could reasonably introduce for a callback-based subscription model. In other languages where the subscriptions are iterator-style, there's a thread to throw the execption into, but not for callback-style.

Open to ideas/discussion, though!